src/Kiota.Builder/Writers/CSharp/CSharpConventionService.cs

```diff
@@ -89,8 +89,15 @@ public override string GetTypeString(CodeTypeBase code, CodeElement targetElemen
         }
         private string TranslateTypeAndAvoidUsingNamespaceSegmentNames(CodeType currentType, CodeElement targetElement)
         {
+            var parentElements = new List<string>();
+            if(targetElement.Parent is CodeClass parentClass)
+                parentElements.AddRange(parentClass.Methods.Select(x => x.Name).Union(parentClass.Properties.Select(x => x.Name)));
```

Comment:
yes but that's a feature, not a bug, keep in mind we're comparing the *type name* of the member against the name of all the members. Not the *name* of the member against the name of all the members.
So effectively if have a member called MailboxSettings of type MailboxSettings, I do want a fully qualified name because otherwise the compiler gets confused when seeing MailboxSettings.CreateFromDiscriminatorValue inthe deserializer callback.

Reply:
Sorry, I wasn't too clear here. I agree. That's true. 
The question really is if we want both the declaration of member to have the fully qualified type name and the parameter in the callback to have then full type name?
Or just the parameter in the deserializer callback?

Reply:
That's fair. I'm happy to go with this.

(Also, apologies for the delayed responses on this PR)